### DISCUSSION TITLE
[SOLVED] Cannot read properties of undefined (reading 'call')

### DISCUSSION THREAD

**Discussion by txfs19260817** [nextauthjs/next-auth#2877] [Help]
## Description 🐜
With version `4.0.0-beta.2`, after adding `SessionProvider` in `_app.tsx` of my Typescript project, this exception raised.

```
Cannot read properties of undefined (reading 'call')
```

Based on my previous experience, I solved it by adding the support of top-level await into `next.config.js`:

```js
webpack: (config) => {
  config.experiments = {
    layers: true,
    topLevelAwait: true
  };
  return config;
},
```

I didn't dig into the root cause. Just take a note here for folks who might enounter the same problem.

## Environment 🖥
OS: Windows 10 
Node: v12.15.0 (maybe I have to upgrade it)
Yarn: 1.22.10
Packages:
    "next": "^v11.1.3-canary.39",
    "next-auth": "4.0.0-beta.2",
    "react": "^17.0.2",

**Comment by balazsorban44** [comment]
topLevelAwait is not necessary for `next-auth`, could you please elaborate? Do you have  a code example that required you that setting?

**Reply by txfs19260817** [reply]
Hi, this is the project and I encountered this problem when I was working on it. https://github.com/txfs19260817/next-chakra-template .
It seems that this might involve the `Link` component from `next/link`.
And yet, I could not recall what drove me to add top-level await to solve it, lol.

**Reply by balazsorban44** [reply]
OK, I just wasn't sure why did you open this discussion in our repo then. 🙂 It doesn't sound related to `next-auth`.
